api/routes.js

```diff
@@ -0,0 +1,176 @@
+      if (session) {
+        var id = session.data.id
+        server.models.users.get({id: id}, function (err, user) {
+          if (err) {
+            response.json({
+              'status': 'warning',
+              'message': err.message
+            }).pipe(res)
+            return
+          }
+          response.json({
+            'status': 'success',
+            'user': user
+          }).pipe(res)
+        })
+      } else {
+        response.json({
+          'status': 'warning',
+          'message': 'No current user.'
+        }).pipe(res)
+      }
+    })
+  })
+
+  router.addRoute('/api/:model/:id?', function(req, res, opts) {
+    res.setHeader('content-type', 'application/json')
+    var id = opts.params.id
+    var model = server.models[opts.params.model]
+    
+    if (!model) {
+      res.statusCode = 400
+      response.json({error: 'Model not found'}).pipe(res)
+      return 
+    }
+    
+    var method = req.method.toLowerCase()
+    var params = {}
+    if (id) params.id = id
+      
+    authorize(server, req, res, opts, function(err) {
+      if (err) return unauthorized(res)
+      
+      if (method === 'get') {
+        var query = getSecondaryQuery(req, model, params)
+        if (query) return secondaryQuery(req, model, query, respond)
+      }
+      
+      model.handler.dispatch(req, params, respond)
+      
+      function respond(err, data) {
+        if (err) {
+          var code = 400
+          if (err.notFound) code = 404
+          if (err.statusCode) code = err.statusCode
+          res.statusCode = code
+          response.json({status: 'error', error: err.message}).pipe(res)
+          return
+        }
+
+        if (!data) data = {status: 'error', error: 'no data returned'}
+      
+        if (method === 'get' || method === 'delete') {
+          res.statusCode = 200
+          response.json(data).pipe(res)
+          return
+        }
+
+        // if put or post then `data` should look like {created: bool, data: row}
+        if (method === 'put' || method === 'post') {
+          res.statusCode = data.created ? 201 : 200
+          // TODO should we treat validation errors as a 200 or a 4xx?
+          if (data.status && data.status === 'error') {
+            res.statusCode = 200
+            response.json(data).pipe(res)
+            return
+          }
+          response.json(data.data).pipe(res)
+          return
+        }
+
+        // should never get here
+        debug('unknown model method', method)
+        res.end()
+      }
+    })
+    
+  })
+
+  return router
+}
+
+function unauthorized(res) {
+  var code = 401
+  res.statusCode = code
+  response.json({status: 'error', error: 'action not allowed'}).pipe(res)
+}
+
+function secondaryQuery(req, model, query, cb) {
+  if (query instanceof Error) return cb(query)
+  var indexer = model.indexes[query.key]
+  debug('finding by', query.key, 'with value', query.value)
+  indexer.findOne(query.value, function(err, id) {
```

Comment:
cool sounds good, I think we'd just need to use https://github.com/mafintosh/level-indexer#stream--indexfindoptions-cb instead of `fineOne` and then format the results correctly as JSON